jwst/extract_1d/extract.py

```diff
@@ -133,31 +133,41 @@ def open_extract1d_ref(refname: str, exptype: str) -> dict:
         handle for the jwst.datamodels object for the extract1d file.
     """
 
+    # the extract1d reference file can be 1 of three types:  'json', 'fits', or  'asdf'
+    refname_type = refname[-4:].lower()
     if refname == "N/A":
         ref_dict = None
     else:
-        fd = open(refname)
-        try:
-            ref_dict = json.load(fd)
-            ref_dict['ref_file_type'] = FILE_TYPE_JSON
-            fd.close()
-        except (UnicodeDecodeError, JSONDecodeError):
-            # input file is not JSON so lets try fits and then asdf
-            fd.close()
+        if refname_type == 'json':
+            fd = open(refname)
             try:
-                fd = fits.open(refname)
-                fits_present = 1
+                ref_dict = json.load(fd)
+                ref_dict['ref_file_type'] = FILE_TYPE_JSON
                 fd.close()
-            except OSError:
-                fits_present = 0
-            if fits_present:
+            except (UnicodeDecodeError, JSONDecodeError):
+                # Input file does not load correctly as json file.
+                # Probably an error in json file
+                fd.close()
+                log.error(f"Extract1d json reference file has an error, run a json validator off line and fix the file")
```

Comment:
Need to remove the "f" formatter, because there aren't any values to format into the string (per flake8). Occurs in a couple other places further down too.